application/core/LSETwigViewRenderer.php

```diff
@@ -812,4 +830,17 @@ public function getLoader()
     {
         return $this->_twig->getLoader();
     }
+
+    /**
+     * Adds custom user extensions
+     * @param array $extensions @see self::$user_extensions
+     */
+    public function addUserExtensions($extensions)
+    {
+        foreach ($extensions as $extName) {
+            Yii::setPathOfAlias('extName_' . $extName, Yii::app()->getConfig('usertwigextensionrootdir') . '/' . $extName . '/');
```

Comment:
No need to filter $extName here ?
It's a public function : then can be used by any plugins

Maybe `Yii::setPathOfAlias` is secure :)

Reply:
I didn't see it in master …

Reply:
Sanitize against what? What kind of sanitization?

Reply:
file/directory trasversal

Reply:
setPathOfAlias didn't sanitize directory : https://github.com/yiisoft/yii/blob/master/framework/YiiBase.php#L391

Then extension dev can set '../../../../etc/passwd' to path of alias (addUserExtensions is a public function, and user_extensions is a public var too.)

Reply:
Yes, we could sanitize that. Not much of a hassle. Not sure it is worth it.
Still, after sanitizing, they could do an suspicious include in the new extension.

@olleharstedt Thoughts?

Reply:
Probably a good idea to filter, BUT: It also depends on the assumed access to even use this feature. If file system access is needed to use the feature, filtering won't matter. Still, yeah, why not.

Reply:
I put discussion … i thihnk we can do it after . Not required update , but seems better in my opionon

Reply:
It still public function.

1. In controller : we must always filter/check access etc on all public function
2. In [twig function](https://github.com/LimeSurvey/LimeSurvey/blob/master/application/core/LS_Twig_Extension.php) : we must filter all data if we include something (can be used by Template manager, remind the issue with [getAllAnswers](https://github.com/LimeSurvey/LimeSurvey/commit/69784616b81594d4fd79b0e018d46d461161cffc) )
3. Here ? Unsure …

Reply:
I think it doesn't worth it.
This is to include files, right?
If someone want's to include a suspicious file, they can include a valid file here and then include a dangerus one on the valid include.

Reply:
Yes : «here ? Unsure … » ;)